### PR TITLE
add ability to block RFC1989 IPs, use community with API key

### PR DESCRIPTION
**Pull request by ntsmith7**
218.201.174.22 

RFC 1918 IPs: 
169.254.255.255  
192.168.255.255

### CODE REVIEW

File: config/config.js [hunk 1]
```diff
@@ -120,18 +120,18 @@ module.exports = {
       adminOnly: false
     },
     {
-      key: 'ignoreIp',
+      key: 'ignoreRC1918Ip',
       name: 'Ignore RFC1989 IPs',
       description: 'Ignore RFC1989 IP addresses',
-      default: true,
+      default: false,
```
Comment (bradchiappetta) [inline config/config.js:126]: This really should default to true, or even just be built in and not a configuration.  We will never have data on RFC1918 IP addresses, there isn't ever a need to query them in our service.

File: integration.js [hunk 2]
```diff
@@ -53,36 +58,29 @@ const useGreynoiseCommunityApi = (entities, options, cb) => {
...
-        json: true
-      };
-
-      // if key exists, call the community api with the provided key.
-      if (options.apiKey) {
-        requestOptions.headers.key = options.apiKey;
-      }
+    let requestOptions = {
+      method: 'GET',
+      uri: 'https://api.greynoise.io/v3/community/' + entity.value,
```
Comment (bradchiappetta) [inline integration.js:63]: You may want to update this to use the base URL provided in the config

`uri: options.url + '/v3/community/' + entity.value,

File: config/config.js (hunk 1 above)
Comment (ntsmith7) [reply config/config.js]: Hey @bradchiappetta  - 

Thanks for catching these. Made the changes.

Comment (bradchiappetta) [reply config/config.js]: No problem, happy to joint effort these, benefits us both in the end.  Would love to get this pushed to my client for testing before we release, if possible.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "main": "./integration.js",
   "name": "GreyNoise",
-  "version": "3.2.0-beta",
+  "version": "3.2.1-beta",
```
Comment (bradchiappetta) [inline package.json:4]: I feel like this update is worthy of a 3.3.0?  Also, any reason this is still -beta?

File: config/config.js [hunk 4]
```diff
@@ -94,7 +94,7 @@ module.exports = {
       key: 'url',
       name: 'GreyNoise Enterprise URL',
       description: 'The URL path to the Enterprise GreyNoise API you wish to use.',
-      default: 'https://enterprise.api.greynoise.io/v2',
+      default: 'https://enterprise.api.greynoise.io',
```
Comment (bradchiappetta) [inline config/config.js:97]: @ntsmith7 : this should be `https://api.greynoise.io`

File: config/config.js [hunk 5]
```diff
@@ -118,6 +118,24 @@ module.exports = {
...
+      name: 'Ignore RFC1989 IPs',
+      description: 'Ignore RFC1989 IP addresses',
+      default: true,
+      type: 'boolean',
+      userCanEdit: true,
+      adminOnly: false
+    },
+    {
+      key: 'standardApi',
+      name: 'Search using the standard API',
```
Comment (bradchiappetta) [inline config/config.js:133]: Would you be open to this update?

```
{
      key: 'subscriptionApi',
      name: 'Search using the Subscription API's,
      description: 'Search using the GreyNoise Subscription Based APIs.  When disabled, the GreyNoise Community API will be used (only supports IP lookups)',
      default: true,
      type: 'boolean',
      userCanEdit: true,
      adminOnly: false
    }

If you are okay with this, you'll need to update the code below.

File: config/config.js [hunk 6]
```diff
@@ -118,6 +118,24 @@ module.exports = {
       type: 'boolean',
       userCanEdit: true,
       adminOnly: false
+    },
+    {
+      key: 'ignoreRC1918Ip',
+      name: 'Ignore RFC1989 IPs',
+      description: 'Ignore RFC1989 IP addresses',
```
Comment (bradchiappetta) [inline config/config.js:125]: Can we expand the description to:

`Prevents RFC1989 IP addresses from being sent to the GreyNoise API for enrichment`

File: package.json (hunk 3 above)
Comment (bradchiappetta) [reply package.json:4]: *bump on this one too*

**Comment by bradchiappetta** [comment]
@ntsmith7 : any thoughts on the open comments / suggestions?

File: config/config.js (hunk 6 above)
Comment (ntsmith7) [reply config/config.js:125]: Hey @bradchiappetta - thanks for these suggestions. I will go ahead and implement them.

File: integration.js [hunk 7]
```diff
@@ -45,10 +45,10 @@ function startup(logger) {
 function doLookup(entities, options, cb) {
   Logger.trace({ entities });
 
-  if(!options.apiKey) {
-    useGreynoiseCommunityApi(entities, options, cb);
-  } else {
+  if (options.standardApi) {
```
Comment (bradchiappetta) [inline integration.js:48]: I think this needs to be updated to match: subscriptionApi

**Comment by ntsmith7** [comment]
@bradchiappetta  - I do not see the code your comments are referring to.

**Comment by bradchiappetta** [comment]
> @bradchiappetta - I do not see the code your comments are referring to.

@ntsmith7 : weird, looks like GH lost all the line numbers with those comments.  They were all tied to templates/gn-block.hbs when I added them.  Happy to review in real-time tomorrow if that would be helpful and you can't align where they go.  You might have to review them in reference this commit: https://github.com/polarityio/greynoise/commit/5131a256609ce779bee5509cb64f9a23b93a3836

File: templates/gn-block.hbs [hunk 8]
```diff
@@ -236,324 +254,324 @@
...
-      <span class='p-key'>IP: </span>
-      <a href='{{details.link}}' class='p-link'>{{details.ip}}
-        {{fa-icon 'external-link-square' class='external-link-icon'}}</a>
+      <span class="p-key">IP: </span>
+      <a href="{{details.link}}" class="p-link">{{details.ip}}
+        {{fa-icon "external-link-square" class="external-link-icon"}}</a>
       |
-      <a class='p-link' href='https://www.virustotal.com/gui/search/{{details.ip}}'>VT
-        {{fa-icon 'external-link-square' class='external-link-icon'}}</a><br />
+      <a class="p-link" href="https://www.virustotal.com/gui/search/{{details.ip}}">VT
```
Comment (bradchiappetta) [inline templates/gn-block.hbs:543]: @ntsmith7 : seems like line 543 still has a Virustotal link?  I think this should be a GN link instead?  Can you confirm?

**Comment by bradchiappetta** [comment]
> We need to add "TRUST LEVEL" in this section as well.

@ntsmith7 : in the gn-block.hbs section where we pull the RIOT full details from the subscription API, can we add the "Trust Level" value as well, please?

**Comment by ntsmith7** [comment]
@bradchiappetta  - 

Per the docs, this is the RIOT response for the subscription API: 

```
{
"ip": "8.8.8.8",
"riot": true,
"category": "public_dns",
"name": "Google Public DNS",
"description": "Google's global domain name system (DNS) resolution service.",
"explanation": "Public DNS services are used as alternatives to ISP's name servers. You may see devices on your network communicating with Google Public DNS over port 53/TCP or 53/UDP to resolve DNS lookups.",
"last_updated": "2020-12-17T20:03:25Z",
"logo_url": "https://www.gstatic.com/devrel-devsite/prod/v9d82702993bc22f782b7874a0f933b5e39c1f0889acab7d1fce0d6deb8e0f63d/cloud/images/cloud-logo.svg",
"reference": "https://developers.google.com/speed/public-dns/docs/isp#alternative"
} 
```


I don't see a trust level. Am I looking at the incorrect response?

**Comment by bradchiappetta** [comment]
@ntsmith7 : looking good, thanks for all the changes.  Can you push this last commit to my local client (or ask Joe to) so I can test locally and see if there is anything else?

**Pull request closed without merging**